wpilibc/src/main/native/cpp/DriverStation.cpp

```diff
@@ -26,41 +28,70 @@
 #include "frc/WPIErrors.h"
 
 namespace frc {
+// A simple class which caches the previous value written to an NT entry
+// Used to prevent redundant, repeated writes of the same value
+template <class T>
+class MatchDataSenderEntry {
+ public:
+  MatchDataSenderEntry(std::shared_ptr<nt::NetworkTable> table,
+                       const std::string& key, const T& initialVal) {
+    static_assert(std::is_convertible<decltype(initialVal), bool>() ||
+                      std::is_convertible<decltype(initialVal), double>() ||
+                      std::is_convertible<decltype(initialVal), std::string>(),
+                  "Invalid type for MatchDataSenderEntry - must be convertable "
+                  "to bool, double or std::string");
```

Comment:
Does this work if `wpi::Twine` is used instead of `std::string`?  Doing so would match the underlying SetString() et al calls.

Reply:
Yes, updated this plus the Set() call below to use wpi::Twine.